code/mob/living/silicon/robot.dm

```diff
@@ -509,6 +509,46 @@
 					playsound(src.loc, 'sound/vox/birdwell.ogg', 50, 1, channel=VOLUME_CHANNEL_EMOTE)
 					message = "<b>[src]</b> birdwells."
 
+			if ("dance")
+				if (src.emote_check(voluntary, 50))
+					var/dancemove = rand(1,4)
+					switch(dancemove)
+						if (1)
+							message = "<B>[src]</B> does the robot."
```

Comment:
Do you think you could make these use `animate` instead?

Reply:
Never touched it before, is it straightforward to recreate the movements?

Also these are taken from humans so those will need refactoring.